src/tdigest_api.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-APACHE for a copy of the license.
+ */
+
+#include "postgres.h"
+
+#include "tdigest.h"
+#include "fmgr.h"
+#include "export.h"
+
+#include "lib/stringinfo.h"
+#include "libpq/pqformat.h"
+#include "nodes/nodeFuncs.h"
+#include "utils/builtins.h"
+#include "utils/bytea.h"
+#include "utils/datum.h"
+#include "utils/typcache.h"
+#include "compat.h"
+
+#define TDIGEST_TYPENAME "tdigest"
+
+/*
+ * For future extensibility and memory management reasons, we use a struct enclosing a TDigest,
+ * rather than a raw TDigest
+ */
+typedef struct TDigestTState
+{
+	TDigest *td;
+} TDigestTState;
+
+/* Transition support functions */
+extern Datum ts_tdigest_print_api(PG_FUNCTION_ARGS);
```

Comment:
The naming of these functions is a bit odd, as it looks like we're e.g. printing a T-Digest API. If we want to have API in the I'd reccommend it be as a prefix, such as `ts_api_tdigest_print` or `ts_tdigest_api_print`. An alternative would be to use `sql` as the suffix.